### PR TITLE
DEV: Convert fast-edit on mobile to a modal

### PR DESCRIPTION
**Pull request by CvX**
Motivation: to fix an issue with fast-edit positioning on mobile (android) Trying to correctly position that textarea/popper element proved difficult

Simulator Screenshot - iPhone SE (3rd generation) - 2023-07-18 at 12 57 40

### CODE REVIEW

File: app/assets/javascripts/discourse/app/components/quote-button.js [hunk 1]
```diff
@@ -361,113 +356,69 @@ export default Component.extend(KeyEnterEscape, {
...
-  ),
-
   @action
   insertQuote() {
     this.attrs.selectText().then(() => this._hideButton());
   },
 
   @action
-  _toggleFastEditForm() {
+  async _toggleFastEditForm() {
```
Comment (davidtaylorhq) [inline app/assets/javascripts/discourse/app/components/quote-button.js:365]: Does this need to be async? I don't see any `await` inside 👀

File: app/assets/javascripts/discourse/app/components/quote-button.js [hunk 2]
```diff
@@ -361,113 +356,69 @@ export default Component.extend(KeyEnterEscape, {
...
-              );
-
-              // ensures we correctly scroll to caret and reloads composer
-              // if we do another selection/edit
-              textarea.blur();
-              textarea.focus();
-            });
-        }
-      );
+      return;
```
Comment (davidtaylorhq) [inline app/assets/javascripts/discourse/app/components/quote-button.js:379]: Unneeded return?

**Review by davidtaylorhq — APPROVED** [review]

File: app/assets/javascripts/discourse/app/components/quote-button.js (hunk 1 above)
Comment (CvX) [reply app/assets/javascripts/discourse/app/components/quote-button.js:365]: both `async` and `return` are needed 😃 the code below this block is also in this action 😛

Comment (davidtaylorhq) [reply app/assets/javascripts/discourse/app/components/quote-button.js:365]: 🤦

**Pull request merged**